optuna/samplers/grid.py

```diff
@@ -0,0 +1,175 @@
+import collections
+import itertools
+import random
+
+from optuna.samplers.base import BaseSampler
+from optuna import type_checking
+
+if type_checking.TYPE_CHECKING:
+    from typing import Any  # NOQA
+    from typing import Dict  # NOQA
+    from typing import List  # NOQA
+    from typing import Union
+
+    from optuna.distributions import BaseDistribution  # NOQA
+    from optuna.structs import FrozenTrial  # NOQA
+    from optuna.study import Study  # NOQA
+
+    GridValueType = Union[str, float, int, bool, None]
+
+
+class GridSampler(BaseSampler):
```

Comment:
I've just remembered that @crcrpar introduced `@experimental()` decorator (https://github.com/optuna/optuna/pull/884).

Reply:
Sounds good. I put the decorator as well.